### PR TITLE
Closes #116

### LINKED ISSUE #116: class `LogisticNormalMixture` clean up
Please edit appropriately below if needed.

To do:

- [x] Revise documentation
  - [x] RStudio outline section dividers (use `# function or class name ----`)
  - [x] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [x] Replace `##'` by `#'`
  - [x] Add to `pkgdown.yml`
  - [x] Remove file history and the timestamp in comments on top of file
- [x] Add unit tests
  - [x] For classes:
    - [x] Test default constructor
    - [x] For reference classes: test methods
  - [x] For functions: as usual
- [x] Refactor as needed
  - [x] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [x] Keep externally used names so that we don't break downstream code
  - [x] Remove `keywords` tags
  - [x] For functions or methods:
    - [x] Add assertions for arguments using `checkmate` functions
    - [x] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [x] Consider renaming internal variables etc. if needed
    - [x] Avoid explicit use of `return`
  - [x] For classes:
    - [x] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [x] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [x] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #116

### CODE REVIEW

**Comment by wwojciech** [comment]
@dinakar29 Hi Dinakar - do you know wht gitleaks / PII Check is failing here? I have never seen this before. Thanks!

**Comment by danielinteractive** [comment]
> @dinakar29 Hi Dinakar - do you know wht gitleaks / PII Check is failing here? I have never seen this before. Thanks!

I understand IDR team with @dinakar29 is currently working on this. But no problem for us since a failure there does not stop us from merging.

File: R/Model-class.R [hunk 1]
```diff
@@ -578,6 +578,129 @@ LogisticKadane <- function(theta, xmin, xmax) {
...
+  ),
+  validity = v_model_logistic_normal_mixture
+)
+
+## constructor ----
+
+#' @rdname LogisticNormalMixture-class
+#'
+#' @param comp1 (`list`)\cr specifications of the first component, i.e. a named list
+#'   with `mean` and `cov` for the first bivariate normal prior mean vector.
```
Comment (danielinteractive) [inline R/Model-class.R:642]: maybe mention here that the precision matrix is then internally calculated

File: R/Model-validity.R [hunk 2]
```diff
@@ -102,3 +102,59 @@ validate_model_logistic_kadane <- function(object) {
...
+    o$check(is_prec_2x2, "prec must be 2x2 matrix without any missing values")
+    if (is_cov_2x2) {
+      o$check(
+        h_is_positive_definite(comp$cov),
+        "cov must be positive-definite matrix"
+      )
+    }
+    if (is_prec_2x2) {
+      o$check(
+        all.equal(comp$cov %*% comp$prec, diag(1, 2), check.attributes = FALSE) == TRUE,
```
Comment (danielinteractive) [inline R/Model-validity.R:135]: nice!

**Review by danielinteractive — APPROVED** [review]
thx Wojciech! nice work!

**Comment by dinakar29** [comment]
> > @dinakar29 Hi Dinakar - do you know wht gitleaks / PII Check is failing here? I have never seen this before. Thanks!
> 
> I understand IDR team with @dinakar29 is currently working on this. But no problem for us since a failure there does not stop us from merging.

Fix here: https://github.com/insightsengineering/presidio-cli/pull/7, will be released in the next iteration of the Action and the CLI.

**Pull request merged**